cachi2/core/package_managers/pip.py

```diff
@@ -88,10 +90,13 @@ def fetch_pip_source(request: Request) -> RequestOutput:
             package.requirements_build_files,
         )
 
-        components.append(Component.from_package_dict(info["package"]))
+        # TODO setup purl in Component / better way
+        purl = _generate_purl(info["package"], request.source_dir)
```

Comment:
we do need a subpath for the main package though (if the main package is not in the root of the source dir)